daml-script/daml/Daml/Script/Error.daml

```diff
@@ -0,0 +1,73 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Script.Error where
+
+import DA.Stack
+
+data SubmitError
```

Comment:
I think eventually we will want an opaque representation plus extractor functions (similar to `AnyTemplate` + one `fromAnyTemplate` per type) but this is a good start and does a good job of collecting the error states